### PR TITLE
NDRS-1129: Index Deploys with Blockhashes

### PR DESCRIPTION
**Pull request by fizyk20**
https://casperlabs.atlassian.net/browse/NDRS-1129

This adds an efficient way of finding a block a deploy hash belongs to.

### CODE REVIEW

File: node/src/components/storage.rs [hunk 1]
```diff
@@ -772,6 +814,19 @@ impl Storage {
...
+    fn get_block_header_by_deploy_hash<Tx: Transaction>(
+        &self,
+        tx: &mut Tx,
+        deploy_hash: DeployHash,
+    ) -> Result<Option<BlockHeader>, LmdbExtError> {
+        self.deploy_hash_index
+            .get(&deploy_hash)
+            .and_then(|block_hash| self.get_single_block_header(tx, block_hash).transpose())
+            .transpose()
+    }
```
Comment (xcthulhu) [inline node/src/components/storage.rs:828]: I don't think this function is necessary.  It should be sufficient to just return a block hash...

**Review by xcthulhu — APPROVED** [review]
LGTM

**Comment by xcthulhu** [comment]
Looks like we just need to get tests passing...

**Comment by fizyk20** [comment]
The failure is just a fluke (socket already in use). A restart of the build should fix it.

**Pull request merged**